micronota/parsers/cmscan.py

```diff
@@ -0,0 +1,428 @@
+            # information always at the same key
+            if program is 'CMSEARCH':
+                attributes['SEQUENCE_NAME'] = fields[0]
+                attributes['SEQUENCE_ACCESSION'] = fields[1]
+                attributes['MODEL_NAME'] = fields[2]
+                attributes['MODEL_ACCESSION'] = fields[3]
+            elif program is 'CMSCAN':
+                attributes['SEQUENCE_NAME'] = fields[2]
+                attributes['SEQUENCE_ACCESSION'] = fields[3]
+                attributes['MODEL_NAME'] = fields[0]
+                attributes['MODEL_ACCESSION'] = fields[1]
+            else:
+                raise CmscanFormatError("Argument 'program' must be either "
+                                        "'CMSEARCH' or 'CMSCAN'!")
+
+            # iterate through all keys that have not already be handled above
+            for key in _orderedKeys:
+                if key in ['SEQUENCE_START_POSITION',
+                           'SEQUENCE_END_POSITION',
+                           'SEQUENCE_NAME',
+                           'SEQUENCE_ACCESSION',
+                           'MODEL_NAME',
+                           'MODEL_ACCESSION']:
+                    continue
+                attributes[key] = fields[_COLUMNS[key]['position']]
+
+            # cmscan works on multiple sequence in one FASTA file. We want to
+            # yield a separate object for each sequence, thus we create a new
+            # one whenever the ID changes
+            if (currentsequence != attributes['SEQUENCE_NAME'] and
+                    currentsequence is not False):
+                yield annotations
+                annotations = {}
+
+            # a metadata interval is made out of 'frozen dictionary' aka
+            # Features object, where the its hashable value constitutes the
+            # key and the value is a list of tuples, aka intervals
+            annotations[Feature(attributes)] = [(hitStart, hitEnd)]
+            # store current sequence id for the next iteration
+            currentsequence = attributes['SEQUENCE_NAME']
+
+    yield annotations
+
+
+def _writeheader(fh):
+    """ Print a header for cmscan hits. It is similar to 'CMscan --tblout' but
+        not identical, since we do not check for longest occurring names, but
+        use fixed field width. """
+
+    # write file header
+    lineCaption = '#'
+    lineDelimeter = '#'
+    for key in _orderedKeys:
+        length = _COLUMNS[key]['width']
+        if _COLUMNS[key]['position'] == 0:
+            length = length-1
+        lineCaption = lineCaption + _printField(
+               _COLUMNS[key], _COLUMNS[key]['caption'], length)
+        lineDelimeter = lineDelimeter + ('-' * length)
+        if _COLUMNS[key]['position']+1 != len(_COLUMNS):
+            lineCaption = lineCaption + " "
+            lineDelimeter = lineDelimeter + " "
+    fh.write(lineCaption + "\n")
+    fh.write(lineDelimeter + "\n")
+
+
+@cmscan.writer(IntervalMetadata)
```

Comment:
as we discussed, we don't need writers. probably remove all associated functions/varaibles?